File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = [1 for _ in nums]
        prefix = nums[0]

        for i in range(1, len(nums)):
            ans[i] = prefix
            prefix *= nums[i]

        postfix = 1
        for i in range(len(nums)-1,-1,-1):
            ans[i] *= postfix
            postfix *= nums[i]
        
        return ans
        

File: Data Structures & Algorithms/subsets/submission-0.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans; 
        solve_recursive(0, nums, ans, {});
        return ans; 
    }

    void solve_recursive(int index, vector<int> &nums, vector<vector<int>> &ans, vector<int> curr)
    {
        ans.push_back(curr);
        for (int i = index; i < nums.size(); i++)
        {
            curr.push_back(nums[i]);
            solve_recursive(i + 1, nums, ans, curr);
            curr.pop_back();
        }
    }
};
